ReactAndroid/src/main/java/com/facebook/react/uimanager/LayoutShadowNode.java

```diff
@@ -763,17 +764,83 @@ public void setBorderWidths(int index, float borderWidth) {
         ViewProps.RIGHT,
         ViewProps.TOP,
         ViewProps.BOTTOM,
+        ViewProps.INSET,
+        ViewProps.INSET_BLOCK,
+        ViewProps.INSET_BLOCK_END,
+        ViewProps.INSET_BLOCK_START,
+        ViewProps.INSET_INLINE,
+        ViewProps.INSET_INLINE_END,
+        ViewProps.INSET_INLINE_START,
       })
   public void setPositionValues(int index, Dynamic position) {
     if (isVirtual()) {
       return;
     }
 
     final int[] POSITION_SPACING_TYPES = {
-      Spacing.START, Spacing.END, Spacing.LEFT, Spacing.RIGHT, Spacing.TOP, Spacing.BOTTOM
+      Spacing.START,
+      Spacing.END,
+      Spacing.LEFT,
+      Spacing.RIGHT,
+      Spacing.TOP,
+      Spacing.BOTTOM,
+      Spacing.ALL,
+      Spacing.BLOCK,
+      Spacing.BLOCK_END,
+      Spacing.BLOCK_START,
+      Spacing.INLINE,
+      Spacing.INLINE_END,
+      Spacing.INLINE_START
     };
 
-    int spacingType = maybeTransformLeftRightToStartEnd(POSITION_SPACING_TYPES[index]);
+    mPosition[POSITION_SPACING_TYPES[index]] = position;
+    updatePositionValues();
+  }
+
+  private void updatePositionValues() {
+    setYogaPosition(Spacing.TOP, mPosition[Spacing.TOP]);
+    setYogaPosition(Spacing.BOTTOM, mPosition[Spacing.BOTTOM]);
+    setYogaPosition(Spacing.START, mPosition[Spacing.START]);
+    setYogaPosition(Spacing.END, mPosition[Spacing.END]);
+
+    if (!I18nUtil.getInstance().doLeftAndRightSwapInRTL(getThemedContext())) {
+      setYogaPosition(Spacing.LEFT, mPosition[Spacing.LEFT]);
+      setYogaPosition(Spacing.RIGHT, mPosition[Spacing.RIGHT]);
+    } else {
+      setYogaPosition(Spacing.START, mPosition[Spacing.LEFT]);
+      setYogaPosition(Spacing.END, mPosition[Spacing.RIGHT]);
+    }
+
+    // Aliases with precedence
+    if (mPosition[Spacing.ALL] != null && !mPosition[Spacing.ALL].isNull()) {
+      setYogaPosition(Spacing.ALL, mPosition[Spacing.ALL]);
+    }
+    if (mPosition[Spacing.BLOCK] != null && !mPosition[Spacing.BLOCK].isNull()) {
+      setYogaPosition(Spacing.VERTICAL, mPosition[Spacing.BLOCK]);
+    }
+    if (mPosition[Spacing.INLINE] != null && !mPosition[Spacing.INLINE].isNull()) {
+      setYogaPosition(Spacing.HORIZONTAL, mPosition[Spacing.INLINE]);
+    }
+    if (mPosition[Spacing.INLINE_END] != null && !mPosition[Spacing.INLINE_END].isNull()) {
+      setYogaPosition(Spacing.END, mPosition[Spacing.INLINE_END]);
+    }
+    if (mPosition[Spacing.INLINE_START] != null && !mPosition[Spacing.INLINE_START].isNull()) {
+      setYogaPosition(Spacing.START, mPosition[Spacing.INLINE_START]);
+    }
+
+    // Aliases without precedence
+    if (mPosition[Spacing.BOTTOM] == null || mPosition[Spacing.BOTTOM].isNull()) {
+      setYogaPosition(Spacing.BOTTOM, mPosition[Spacing.BLOCK_END]);
+    }
+    if (mPosition[Spacing.TOP] == null || mPosition[Spacing.TOP].isNull()) {
+      setYogaPosition(Spacing.TOP, mPosition[Spacing.BLOCK_START]);
+    }
+  }
+
+  private void setYogaPosition(int spacingType, Dynamic position) {
```

Comment:
I'm looking closer at the impl of this, and `spacingType` is translated directly to a Yoga Edge.

We already added some more different values to `Spacing`, but since these can only accept Yoga Edge parameters, is it possible to use a more constrained type? E.g. changing sinks which expect a valid Yoga enum to instead use `YogaEdge`?